remote-query/remote-query-server/src/main/java/org/infinispan/query/remote/impl/ProtobufMetadataManagerInterceptor.java

```diff
@@ -276,9 +285,32 @@ private void handlePutKeyValueResult(InvocationContext rCtx, PutKeyValueCommand
          }
 
          if (progressCallback != null) {
-            updateGlobalErrors(rCtx, progressCallback.getErrorFiles(), flagsBitSet);
+            CompletionStage<Void> schemaUpdate = updateSchema(rCtx, progressCallback);
```

Comment:
IMO having a separate `if` here makes sense only if you move this line out of the `if` below:

```
registerProtoFile((String) key, (String) value, progressCallback);
```

I'd rather move the lines of this `if` block inside the `if` block above that assigns a non-null value to `progressCallback` instead.

Reply:
Sure. Also can remove the `else` block as well.